README.md

```diff
@@ -503,110 +574,171 @@ Values have the following effects:
 Defaults to `FromImage`.
 
-#####`os_disk_vhd_container_name`
+##### `os_disk_vhd_container_name`
 The vhd container name is used to create the vhd uri of the virtual machine.
 
 This will transpose with resource_group and the os_disk_vhd_name to become the URI of your virtual hard disk image.
 ~~~
 https://#{resource_group}.blob.core.windows.net/#{os_disk_vhd_container_name}/#{os_disk_vhd_name}.vhd
 ~~~
 
-#####`os_disk_vhd_name`
+##### `os_disk_vhd_name`
 The name of the vhd that forms the vhd URI for the virtual machine.
 
-#####`dns_domain_name`
+##### `dns_domain_name`
 The DNS domain name that to be associated with the virtual machine.
 
-#####`dns_servers`
+##### `dns_servers`
 The DNS servers to be setup on the virtual machine.
 Defaults to `10.1.1.1 10.1.2.4`
 
-#####`public_ip_allocation_method`
+##### `public_ip_allocation_method`
 The public ip allocation method [Static, Dynamic]
 Defaults to `Dynamic`.
 
-#####`public_ip_address_name`
+##### `public_ip_address_name`
 The key name of the public ip address.
 
-#####`virtual_network_name`
+##### `virtual_network_name`
 The key name of the virtual network for the virtual machine. [Virtual Network setup](https://msdn.microsoft.com/en-us/library/azure/jj157100.aspx)
 
-#####`virtual_network_address_space`
+##### `virtual_network_address_space`
 The ip range for the private virtual network. [Virtual Network setup](https://msdn.microsoft.com/en-us/library/azure/jj157100.aspx)
 Default's to `10.0.0.0/16`.
 
-#####`subnet_name`
+##### `subnet_name`
 The private subnet name for the virtual network. [Virtual Network setup](https://msdn.microsoft.com/en-us/library/azure/jj157100.aspx)
 
-#####`subnet_address_prefix`
+##### `subnet_address_prefix`
 Details of the prefix are availabe at [Virtual Network setup](https://msdn.microsoft.com/en-us/library/azure/jj157100.aspx)
 Default's to `10.0.2.0/24`.
 
-#####`ip_configuration_name`
+##### `ip_configuration_name`
 The key name of the ip configuration for the VM.
 
-#####`private_ip_allocation_method`
+##### `private_ip_allocation_method`
 The private ip allocation method [Static, Dynamic]
 Defaults to `Dynamic`.
 
-#####`network_interface_name`
+##### `network_interface_name`
 The Network Interface Controller (nic) name for the virtual machine.
 
-##Known Issues
+##### `extensions`
+The extension to configure on the VM. Supports the following parameters:
+
+##### `publisher`
+The name of the publisher of the extension.
+
+##### `type`
+The type of the extension (e.g. CustomScriptExtension).
+
+##### `type_handler_version`
+The version of the extension to use.
+
+##### `settings`
+The settings specific to an extension (e.g. CommandsToExecute).
+
+##### `protected_settings`
+The settings specific to an extension that are encrypted before passing to the VM.
+
+##### `auto_upgrade_minor_version`
+Indicates whether extension should automatically upgrade to latest minor version.
```

Comment:
These six parameters are keys of an extension but not indented another level so appear to be top-level parameters.

Perhaps there should be some explanation of how extensions works, that it accepts a hash of extension names with a value of either `absent` to uninstall the extension or a hash of these six parameters to install the extension?